lib/ansible/playbook/role/__init__.py

```diff
@@ -281,6 +321,31 @@ def _load_dependencies(self):
 
         return deps
 
+    def _create_arg_spec_validation_task_data(self, argument_spec, argument_spec_name,
+                                              role_name, role_path, role_params):
+        # If the arg spec provides a description, use it to flesh out the validation task name
+        task_name_parts = ["Validating arguments against arg spec '%s'" % argument_spec_name]
+        if 'description' in argument_spec:
+            task_name_parts.append(argument_spec['description'])
+        task_name = ' - '.join(task_name_parts)
+
+        arg_spec_task = {'action': {'module': 'validate_arg_spec',
+                                    'argument_spec': argument_spec,
+                                    'provided_arguments': role_params,
+                                    'validate_args_context': {'type': 'role',
+                                                              'name': role_name,
+                                                              'argument_spec_name': argument_spec_name,
+                                                              'path': role_path},
+                                    },
+                         'name': task_name,
+                         # TODO: use ignore_error from include_role for the validate task?
+                         # 'ignore_errors': ignore_errors,
+                         # 'vars': {'argument_spec': []},
+                         # 'async_val': async_val,
+                         # 'poll': poll},
+                         }
```

Comment:
yes, we should inherit all properties ... not sure if we'll eventually want a 'ignore or skip spec validation' option for roles

Reply:
I'll remove the TODO, since at the moment it inherits the properties.